strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/configuration/ServerConfigurationController.java

```diff
@@ -148,6 +155,29 @@ public ResponseEntity getPort(@RequestHeader(HttpHeaders.ACCEPT) String accept)
         return ResponseEntity.ok(getPortEntityBody(configurationManagementService.getPort(), accept));
     }
 
+    @ApiOperation(value = "Sets the server settings of the service.")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = SUCCESSFUL_SAVE_SERVER_SETTINGS),
+                            @ApiResponse(code = 400, message = FAILED_SAVE_SERVER_SETTINGS) })
+    @PreAuthorize("hasAnyAuthority('CONFIGURATION_SET_BASE_URL', 'CONFIGURATION_SET_PORT')")
+    @PostMapping(value = "/serverSettings",
+                 consumes = MediaType.APPLICATION_JSON_VALUE,
+                 produces = { MediaType.TEXT_PLAIN_VALUE,
+                              MediaType.APPLICATION_JSON_VALUE })
+    public ResponseEntity setServerSettings(@RequestBody @Validated ServerSettingsForm serverSettingsForm,
+                                            BindingResult bindingResult,
+                                            @RequestHeader(HttpHeaders.ACCEPT) String acceptHeader)
+    {
+        if (bindingResult.hasErrors())
+        {
+            throw new RequestBodyValidationException(FAILED_SAVE_SERVER_SETTINGS, bindingResult);
+        }
+
+        configurationManagementService.setBaseUrl(serverSettingsForm.getBaseUrl());
+        configurationManagementService.setPort(serverSettingsForm.getPort());
+
+        return ResponseEntity.ok(getResponseEntityBody(SUCCESSFUL_SAVE_SERVER_SETTINGS, acceptHeader));
```

Comment:
This should actually be `return getSuccessfulResponseEntity(SUCCESSFUL_SAVE_SERVER_SETTINGS, acceptHeader)` :)